cmd/root.go

```diff
@@ -39,34 +39,62 @@ const (
 	apmIndexPattern = "apm"
 )
 
-var libbeatConfigOverrides = common.MustNewConfigFrom(map[string]interface{}{
-	"logging": map[string]interface{}{
-		"metrics": map[string]interface{}{
-			"enabled": false,
-		},
-		"files": map[string]interface{}{
-			"rotateeverybytes": 10 * 1024 * 1024,
-		},
+var libbeatConfigOverrides = []cfgfile.ConditionalOverride{{
+	Check: func(_ *common.Config) bool {
+		return true
 	},
-	"setup": map[string]interface{}{
-		"template": map[string]interface{}{
-			"settings": map[string]interface{}{
-				"index": map[string]interface{}{
-					"codec": "best_compression",
-					"mapping": map[string]interface{}{
-						"total_fields": map[string]int{
-							"limit": 2000,
+	Config: common.MustNewConfigFrom(map[string]interface{}{
+		"logging": map[string]interface{}{
+			"metrics": map[string]interface{}{
+				"enabled": false,
+			},
+			"files": map[string]interface{}{
+				"rotateeverybytes": 10 * 1024 * 1024,
+			},
+		},
+		"setup": map[string]interface{}{
+			"template": map[string]interface{}{
+				"settings": map[string]interface{}{
+					"index": map[string]interface{}{
+						"codec": "best_compression",
+						"mapping": map[string]interface{}{
+							"total_fields": map[string]int{
+								"limit": 2000,
+							},
 						},
+						"number_of_shards": 1,
+					},
+					"_source": map[string]interface{}{
+						"enabled": true,
 					},
-					"number_of_shards": 1,
-				},
-				"_source": map[string]interface{}{
-					"enabled": true,
 				},
 			},
 		},
+	}),
+},
+	{
+		Check: func(cfg *common.Config) bool {
```

Comment:
Can you please add a TODO here to update libbeat, adding a more fit for purpose API?